timesketch/lib/analyzers/feature_plugins/winevt.py

```diff
@@ -0,0 +1,172 @@
+"""This file contains the plugin for Windows event logs feature extraction."""
+
+from typing import List
+
+import logging
+
+from timesketch.lib.analyzers import interface as base_interface
+from timesketch.lib.analyzers.feature_plugins import interface
+from timesketch.lib.analyzers.feature_plugins import manager
+
+logger = logging.getLogger("timesketch.analyzers.feature")
+
+
+class WindowsEventFeatureExtractionPlugin(interface.BaseFeatureExtractionPlugin):
+    """A plugin for Windows event log feature extraction."""
+
+    NAME = "winevt_feature_extraction"
+    DISPLAY_NAME = "Windows Event Log Feature Extraction"
+    DESCRIPTION = (
+        "This plugin extracts Windows event logs attributes from plaso output"
+        " attribute `strings`"
+    )
+
+    EVENT_FIELDS = ["strings"]
+
+    def run_plugin(self) -> str:
+        """Initializes plugin class object.
+
+        Returns:
+            str: Summary of plugin output.
+        """
+
+        # Stores extraction summary for each feature.
+        results = []
+
+        feature_kwargs = self.load_feature_config()
+        if not feature_kwargs:
+            return f"No feature configuration data for {self.NAME}"
+
+        for feature in feature_kwargs:
+            # NOTE: feature_name and feature_config value checking is done at the sink
+            # function `extract_features`.
+            feature_name = feature.get("feature", None)
+            feature_config = feature.get("feature_config", None)
+
+            result = self.extract_features(feature_name, feature_config)
+            if not result:
+                logger.debug("No result for plugin run for %s", feature_name)
+            results.append(result)
+
+        return "\n".join(results)
+
+    def extract_features(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Features extraction name.
+            config (dict): A dict that contains the configuration fo the
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+
+        if not name:
+            raise ValueError("Feature name is empty")
+
+        if not config:
+            raise ValueError("Feature configuration value is empty")
+
+        source_name = config.get("source_name", None)
+        if not source_name:
+            raise ValueError("source_name is empyt")
+
+        if not isinstance(source_name, list):
+            raise TypeError(
+                f"Expecting source_name as a list in {name} and got {type(source_name)}"
+            )
+
+        source_name = source_name[0]
+
+        # Using -1 as a default value for event_identifier and event_version as
+        # 0 is considered as None.
+        event_identifier = int(config.get("event_identifier", -1))
+        event_version = int(config.get("event_version", -1))
```

Comment:
1. Why do we need default values here? The `event_version` field is mandatory, so if it is not set the extraction should be skipped. The `event_identifier` is defined as "optional" but we should maybe revisit if this makes sense.

2. Using `-1` as a 0 adds an extra level of abstraction that is not necessary in my opinion. When using `int(config.get("event_version"))` and event_version is defined as 0, it should work as well. We just need to ensure to check for literal `int(0)` in the later checks to prevent interpretation as None.

Reply:
Removed default value for `event_identifier` and `event_version`. Both `event_identifier` and `event_version` are required and will throw a `TypeError` or `ValueError` depending on the situation.

Since, the feature extractions are registered as a new celery task, it will only affect the feature extraction that is missing the fields or have invalid value type.